cmd/micro/buffer.go

```diff
@@ -73,23 +81,29 @@ type SerializedBuffer struct {
 	ModTime      time.Time
 }
 
+// NewBufferFromString creates a new buffer from a given string with a given path
 func NewBufferFromString(text, path string) *Buffer {
-	return NewBuffer(strings.NewReader(text), int64(len(text)), path)
+	return NewBufferWithPassword(strings.NewReader(text), int64(len(text)), path, "", false)
 }
 
 // NewBuffer creates a new buffer from a given reader with a given path
 func NewBuffer(reader io.Reader, size int64, path string) *Buffer {
-	if path != "" {
-		for _, tab := range tabs {
-			for _, view := range tab.views {
-				if view.Buf.Path == path {
-					return view.Buf
-				}
-			}
-		}
+	return NewBufferWithPassword(reader, size, path, "", false)
+}
+
+// NewBufferWithPassword creates a new buffer from a given reader with a given path
+// and password
+func NewBufferWithPassword(reader io.Reader, size int64, path, password string, passwordPrompted bool) *Buffer {
+	if buf := FindBuffer(path); buf != nil {
+		return buf
 	}
 
 	b := new(Buffer)
+	var err error
+	reader, err = b.Decode(reader, path, password)
+	if err != nil {
+		return NewBufferFromString(fmt.Sprintf("%s: %v", path, err), "")
```

Comment:
Instead of just opening an empty file with the error message, I think micro should print the message and then open the encrypted file (without decrypting it). So this line should probably be something like

```go
TermMessage(fmt.Sprintf("Error: %s: %v", path, err))
return NewBuffer(freader, size, path)
```

I have renamed the argument `reader` to `freader` to avoid the reassignment of the variable.